src/collector/ref_cnt.rs

```diff
@@ -0,0 +1,92 @@
+use std::sync::atomic::{AtomicI64, Ordering};
+
+#[derive(Debug)]
+pub struct GcRefCount {
+    // `total_handles` = count_positive + count_negative
+    // `count_positive` is always >= the actual count >= 0
+    // At the beginning of every collection we recalculate `count_positive`
+    count_positive: AtomicI64,
+    // `count_negative` is always <= 0
+    count_negative: AtomicI64,
```

Comment:
Can you clarify this?

Reply:
Not a specific enough question. Basically this is a complex way of letting `count_positive` always be equal to or greater than the real count .  The reason why it is split is so that threads can decrement without using a lock